File: MATLAB/core_functions/dysco_norm.m
function norm = dysco_norm(eigenvalues,what_norm)

% Giuseppe de Alteriis Oct 2023. With the vector of eigenvalues of the
% matrix I can compute norm very simply

if what_norm==1
    norm = sum(abs(eigenvalues));
elseif what_norm==2 % this should be equal to the sum of all squared elements of the matrix
    norm = sqrt(sum(eigenvalues.^2));
elseif what_norm==Inf
    norm = max(eigenvalues);
end


end

File: Python/core_functions/dysco_norm.py
import numpy as np


def dysco_norm(eigenvalues, what_norm):
    if what_norm == 1:
        norm = np.sum(np.abs(eigenvalues),axis=0)
    elif what_norm == 2:
        norm = np.sqrt(np.sum(eigenvalues ** 2,axis=0))
        # norm = np.sum(eigenvalues ** 2)
    elif what_norm == np.inf:
        norm = np.max(eigenvalues,axis=0)

    return norm